src/app/getting-started/services/provider.service.ts

```diff
@@ -31,8 +36,42 @@ export class ProviderService {
    *
    * @param redirect URL to be redirected to after successful account linking
    */
-  linkGitHub(redirect: string): void {
-    this.link("github", redirect);
+  linkGitHub(redirect: string): void{
+    let headers = new Headers();
+    let githubLinkURL = this.linkUrl + `redirect=` + redirect ;
+
+    this.http
+    .get(this.linkUrl, { headers: headers })
+    .map(response => {
+      let redirectInfo = response.json() as Link;
+      this.redirectToAuth(redirectInfo.redirect);
+    })
+    .catch((error) => {
+      return this.handleError(error);
+    });
+  }
+
+   /**
+   * Link a GitHub and OpenShift Online account to the user account
+   *
+   * @param redirect URL to be redirected to after successful account linking
+   */
+  linkGitHubAndOpenShift(redirect: string): void {
+
+    let parsedToken = jwt_decode(this.auth.getToken());
+    // the new url is /api/link/session 
+    let opensShiftLinkingRedirectUrl = this.getLegacyLinkingUrl("openshift-v3",redirect);
+    this.linkGitHub(opensShiftLinkingRedirectUrl);
+  }
+
+  getLegacyLinkingUrl(provider: string, redirect: string): string{
+    let parsedToken = jwt_decode(this.auth.getToken());    
+    let url = this.loginUrl + `/session?`
+    + "clientSession=" + parsedToken.client_session
+    + "&sessionState=" + parsedToken.session_state
+    + "&redirect=" + redirect // brings us back to Getting Started.
+    + "&provider="+provider;
+    return url
```

Comment:
There are a number of linting issues with this function - please end line 74 with a semicolon at a minimum.  It is good to also space consistently (see the provider assignment on line 73 and the function call only 63).  All of those things help lead to code that is readable and will not break our linting rules once Joshua turns them on globally.